### ISSUE TITLE
negloglik_loss问题请教

### ISSUE THREAD

**Issue by yixinkai123** [XXnormal/RAPIER#8]
input = batch.float().cuda()
            out = model.forward(input)
            mu, logp = torch.chunk(out, 2, dim=1)
            u = (input - mu) * torch.exp(0.5 * logp).cuda()

            negloglik_loss = 0.5 * (u ** 2).sum(dim=1)
            negloglik_loss += 0.5 * input.shape[1] * np.log(2 * math.pi)
            negloglik_loss -= 0.5 * torch.sum(logp, dim=1)

            negloglik_loss = torch.mean(negloglik_loss) 
您好，前辈，请问这里的这个损失函数设计有参考吗？不太理解这里的代码实现，为什么out 经过made模型处理后是128*64，然后chunk切片成两个128*32,然后计算负对数似然损失？

**Comment by XXnormal** [comment]
这个可以参考made以及其实数域扩展实现。实际上是优化了一个核函数，使其拟合所有已有样本
---- 回复的原邮件 ----
***@***.***>发送日期2024年11月09日 19:04 ***@***.***> ***@***.***>主题[XXnormal/RAPIER] negloglik_loss问题请教 (Issue #8)
input = batch.float().cuda() out = model.forward(input) mu, logp = torch.chunk(out, 2, dim=1) u = (input - mu) * torch.exp(0.5 * logp).cuda() negloglik_loss = 0.5 * (u ** 2).sum(dim=1) negloglik_loss += 0.5 * input.shape[1] * np.log(2 * math.pi) negloglik_loss -= 0.5 * torch.sum(logp, dim=1) negloglik_loss = torch.mean(negloglik_loss) 
您好，前辈，请问这里的这个损失函数设计有参考吗？不太理解这里的代码实现，为什么out 经过made模型处理后是12864，然后chunk切片成两个12832,然后计算负对数似然损失？
